### PR TITLE
Resolves #528 and #522

### LINKED ISSUE #528: Task template is showing items in quite a different order than what requester posted
image

This template shows items in a different order than posted. Description is showing at end while it was the first item added.

### PR DESCRIPTION
**Pull request by shirishgoyal**
Template item position is now calculated during submission.
Streamlines template item code
Adds icons to components
Preview now scrolls

image

image

### CODE REVIEW

File: static/js/template/services/template.service.js [hunk 1]
```diff
@@ -59,9 +58,9 @@
         html = '<md-select class="template-item" ng-model="item.answer" layout="' + item.layout + '">' +
             '<md-option ng-repeat="option in item.values.split(\',\')" value="{{option}}">{{option}}</md-option></md-select>';
       } else if (item.type === 'labeled_input') {
-        html = '<div layout="row" style="word-wrap:break-word"><' + item.sub_type + ' flex="90" layout="column">' + item.values + '</' + 
-                item.sub_type + '><md-input-container flex="10" layout="column">' +
-                '<label>rank</label>' +
+        html = '<div layout="row" style="word-wrap:break-word"><' + item.sub_type + ' flex="75" layout="column">' + item.values + '</' +
+                item.sub_type + '><md-input-container flex="25" layout="column">' +
+                '<label>Enter text here</label>' +
```
Comment (dmorina) [inline static/js/template/services/template.service.js]: 25 is too big -> 10 was fine

Comment (shirishgoyal) [inline static/js/template/services/template.service.js]: If we are calling it Labeled Input, then it has to be generic and should allow user to pass any text , so it must have some tangible length.

**Comment by dmorina** [comment]
list icon is always centered so it's a row with layout-alignment="start center" or margin auto top/bottom

**Comment by dmorina** [comment]
and resolve conflicts

**Comment by shirishgoyal** [comment]
Conflicts resolved and audio container merged

**Comment by dmorina** [comment]
generally lgtm

File: static/js/project/controllers/project.controller.js [hunk 2]
```diff
@@ -133,6 +133,13 @@
 
         self.currentProject.categories = [self.currentProject.category];
 
+        var items = self.currentProject.template.items.map(function(item,index){
```
Comment (aginzberg) [inline static/js/project/controllers/project.controller.js]: much better. thank you

**Comment by dmorina** [comment]
@aginzberg is that an lgtm Im hearing? :D

**Comment by aginzberg** [comment]
lol. yes lgtm

**Pull request merged**